src/foundry/client-esm/applications/api/dialog.d.mts

```diff
@@ -307,4 +298,16 @@ type InferButtonReturnTypes<Options extends { buttons?: DialogV2.Button<any>[] }
       ? Callback | string
       : never;
 
+type InferConfirmReturnTypes<
+  Options extends { yes?: InexactPartial<DialogV2.Button<any>>; no?: InexactPartial<DialogV2.Button<any>> },
+> =
+  | (Options["yes"] extends { callback: DialogV2.ButtonCallback<infer YesReturn> } ? YesReturn : true)
+  | (Options["no"] extends { callback: DialogV2.ButtonCallback<infer NoReturn> } ? NoReturn : false);
+
+type InferPromptReturnType<Options extends { ok?: InexactPartial<DialogV2.Button<any>> }> = Options["ok"] extends {
+  callback: DialogV2.ButtonCallback<infer OKReturn>;
+}
+  ? OKReturn
+  : string;
```

Comment:
This also won't distribute. I think something like `Options extends PromptConfig<infer Ok> ? Ok : string` would probably do.